app/brave_main_delegate.cc

```diff
@@ -235,3 +248,40 @@ bool BraveMainDelegate::BasicStartupComplete(int* exit_code) {
 
   return ret;
 }
+
+#if defined(OS_ANDROID)
+void BraveMainDelegate::AdjustSyncServiceUrlForAndroid(
+    std::string* brave_sync_service_url) {
+  DCHECK_NE(brave_sync_service_url, nullptr);
+  const char kProcessTypeHost[] = "host";
+  const char kProcessTypeSwitchName[] = "type";
+
+  // On Android we can detect data dir only on host process, and we cannot
+  // for example on renderer or gpu-process, because JNI is not initialized
+  // And no sense to override sync service url for them in anyway
+  std::string process_type = kProcessTypeHost;
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
```

Comment:
we can simply check `!HasSwitch(switches::kProcessType)` to identify browser process

Reply:
Thanks, I'll modify. Initially I took this logic from https://cs.chromium.org/chromium/src/remoting/host/host_main.cc?type=cs&q=src/remoting/host/host_main.cc&g=0&l=195 , but I can simplify it.

Reply:
@darkdh 
Fixed in https://github.com/brave/brave-core/pull/6288/commits/e23bc503877199fae4b7599cbe040962e4c26dc5 .